frontend/src/app/http-interceptor.ts

```diff
@@ -1,16 +1,32 @@
 import { Injectable } from '@angular/core';
-import { HttpRequest, HttpInterceptor as AngularHttpInterceptor, HttpHandler, HttpEvent } from '@angular/common/http';
+import { HttpRequest, HttpInterceptor, HttpHandler, HttpEvent, HttpXsrfTokenExtractor } from '@angular/common/http';
 import { Observable } from 'rxjs/Observable';
 import { environment } from '../environments/environment';
 
+
 @Injectable()
-export class HttpInterceptor implements AngularHttpInterceptor {
+export class HttpWithCredentialsInterceptor implements HttpInterceptor {
 
   intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
     let authReq = req;
-    if (req.url.indexOf(environment.apiRoot) === 0) {
+    if (req.url.startsWith(environment.apiRoot)) {
       authReq = req.clone({ withCredentials: true });
     }
     return next.handle(authReq);
   }
 }
+
+@Injectable()
+export class HttpXsrfInterceptor implements HttpInterceptor {
+
+  constructor(private tokenService: HttpXsrfTokenExtractor) { }
+
+  intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+    const headerName = 'x-csrftoken';
+    const token = this.tokenService.getToken() as string;
+    if (token !== null && !req.headers.has(headerName) && req.url.startsWith(environment.apiRoot)) {
```

Comment:
I guess it is too late since PR is already merged, however instead of "token !=== null && ..." you could just write 'token && ...'. 
Also, I do not understand why we need '!req.headers.has(headerName)' condition. Is it possible that this header will be in a request before the interceptor catches it, and if yes, where does it come from?

Reply:
Generally this piece of code is a mix of:
 * https://github.com/angular/angular/blob/54e02449549448ebab6f255f2da0b4396665c6f0/packages/common/http/src/xsrf.ts#L75
 * https://github.com/angular/angular/issues/18859#issuecomment-343386182
 * https://stackoverflow.com/a/47591912

That was pretty quick fix, as our dev.volontulo.pl environment response only with `Error`.

If You wish, feel welcome to improve this (I have a plan to write tests once, but there are right now a lot of more important things in my opinion)